File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer>ans=new ArrayList<>();
        if(root==null){
            return 0;
        }
        inorder(root,ans);
        return ans.get(k-1);
        
    }
    public void inorder(TreeNode root,List<Integer>ans){
        if(root==null){
            return;
        }
        if(root.left!=null){
            inorder(root.left,ans);
        }
        ans.add(root.val);
        if(root.right!=null){
            inorder(root.right,ans);
        }
    }
}

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        res=[]
        self.solve(res,root)
        return res[k-1]
        
    def solve(self,res,root):
        if root is None:
            return 
        if root.left:
            self.solve(res,root.left)
        res.append(root.val)
        if root.right:
            self.solve(res,root.right)